tests/integration/test_application.py

```diff
@@ -167,6 +171,48 @@ async def test_upgrade_charm_channel(event_loop):
         assert app.data['charm-url'] != 'cs:ubuntu-0'
 
 
+@base.bootstrapped
+@pytest.mark.asyncio
+async def test_upgrade_charm_switch_channel(event_loop):
+    # Note for future:
+    # This test requires a charm that has different
+    # revisions in channels 'stable' and 'candidate'.
+    # Currently, we use mongodb, but eventually
```

Comment:
You can use the juju-qa-test charm for this. There are 2 tracks currently in use for it. Using that charm would also allow to disambiguation of the channels "stable", "candidate" etc. 

Suggest also adding a track to the risk for the channel. Risk only channels are more typical of charmstore charms.